packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -735,9 +732,7 @@ class SignInStateMachine extends AuthStateMachine<SignInEvent, SignInState> {
     if (authenticationResult != null) {
       final accessToken = await _saveAuthResult(authenticationResult);
       final newDeviceMetadata = authenticationResult.newDeviceMetadata;
-      if (newDeviceMetadata != null &&
-          // ConfirmDevice API requires an identity pool
-          hasIdentityPool) {
```

Comment:
Q: no need to check for `identityPoolConfig != null` anymore?

Reply:
Nope not needed. If there's no identity pool, it will throw an exception later but that's okay. It's handled by the new changes to the Cognito SDK allowing optional auth.